internal/cmd/start.go

```diff
@@ -23,16 +26,51 @@ func SetupStartCli() *cobra.Command {
 }
 
 func startCmd(cmd *cobra.Command, args []string) {
-	proxyURL := makeProxyURL(port)
-	profile.UpdateGlobalEnvironmentVariables(proxyURL)
-	update := fmt.Sprintf("Listen\t%v", port)
-	if err := cntlm.UpdateFile(update); err != nil {
+	cfg, err := config.LoadConfiguration(proxyProfile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if cfg.Proxy.Running {
+		// TODO: A user may want to start another CNTLM instance, logic should handle if its intended or not
+		log.Fatalf("CNTLM Proxy is already running")
+	}
+	proxyURL, err := url.Parse(fmt.Sprintf("http://%s:%v", cfg.Proxy.Address, port))
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = env.UpdateGlobalEnvironmentVariables(proxyURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+	cfg.Proxy.Port = port
+	viper.Set("Proxy.Port", port)
+	viper.Set("Proxy.Running", true)
```

Comment:
I wouldn't use Proxy.Running true/false to determine if something is running, because CNTLM may crash or it may be killed by something that this proxy tool is not aware of.

A more reliable way to determine if CNTLM is running is to store the PID/starttime of the CNtLM process after it's started, and then you can use a syscall to determine whether it is still running.